File: src/main/java/com/yaskovdev/motivated/stay/Task.java
package com.yaskovdev.motivated.stay;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@Data
@Document
class Task {

    @Id
    private String id;

    @NotBlank
    private String userId;

    private String name;

    private String description;

    private LocalDateTime instantOfCreation;

    private LocalDateTime dueDate;

    private boolean closed;
}

File: src/main/kotlin/org/motivepick/domain/Task.kt
package org.motivepick.domain

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.DBRef
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime
import javax.validation.constraints.NotBlank

@Document
class Task {

    @Id
    var id: String? = null
    @NotBlank
    var userId: String? = null
    var name: String? = null
    var description: String? = null
    var instantOfCreation: LocalDateTime? = null
    var dueDate: LocalDateTime? = null
    var closed: Boolean = false

    @DBRef(lazy = true)
    var subtasks: MutableList<Task> = ArrayList()
}
